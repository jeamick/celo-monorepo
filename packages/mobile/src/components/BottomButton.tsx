import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { colors, fontSizes } from 'src/utils/styles'

interface BottomButtonProps {
  buttonMessage: string
  onPress: () => void
}

export default class BottomButton extends React.Component<BottomButtonProps> {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <View style={styles.submitButtonContainer}>
          <Text style={styles.submitButtonText}> {this.props.buttonMessage} </Text>
        </View>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  submitButtonContainer: {
    backgroundColor: colors.GREEN,
    height: 64,
    alignContent: 'center',
    justifyContent: 'center',
  },
  submitButtonText: {
    fontSize: fontSizes.bodyTextSize,
    color: 'white',
    flexDirection: 'column',
    alignItems: 'flex-end',
    textAlign: 'center',
  },
})
