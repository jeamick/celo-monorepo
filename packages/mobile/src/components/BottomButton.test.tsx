import * as React from 'react'
import { TouchableOpacity } from 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import * as renderer from 'react-test-renderer'
import BottomButton from 'src/components/BottomButton'

it('renders correctly', () => {
  const onPress = () => {
    return
  }
  const tree = renderer.create(<BottomButton buttonMessage="test" onPress={onPress} />)
  expect(tree).toMatchSnapshot()
})

it('should call onPress', () => {
  const onPress = jest.fn()
  const wrapper = render(<BottomButton buttonMessage="test" onPress={onPress} />)

  wrapper.getAllByType(TouchableOpacity).forEach((child) => {
    fireEvent.press(child)
  })

  expect(onPress.mock.calls.length).toBe(1)
})
