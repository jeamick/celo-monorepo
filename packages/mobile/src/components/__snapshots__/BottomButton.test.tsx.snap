// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "center",
        "backgroundColor": "#45CD86",
        "height": 64,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "white",
          "flexDirection": "column",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
       
      test
       
    </Text>
  </View>
</View>
`;
